src/search/repositories/search.repository.ts

```diff
@@ -1,4 +1,38 @@
 import { Injectable } from '@nestjs/common';
+import { Board } from 'src/boards/entities/board.entity';
+import { User } from 'src/users/entities/user.entity';
+import { EntityManager } from 'typeorm';
 
 @Injectable()
-export class SearchRepository {}
+export class SearchRepository {
+  constructor(private entityManager: EntityManager) {}
+  async searchBoardsByHeadOrBody(searchQuery: string) {
+    const boardRepository = this.entityManager.getRepository(Board);
+
+    return boardRepository
+      .createQueryBuilder('board')
+      .select()
+      .leftJoinAndSelect('board.user', 'user')
+      .leftJoinAndSelect('user.userImage', 'userImage')
+      .leftJoinAndSelect('board.boardImages', 'boardImages')
```

Comment:
```suggestion
      .leftJoinAndSelect('board.user', 'user')
      .leftJoinAndSelect('user.userImage', 'userImage')
      .leftJoinAndSelect('board.boardImages', 'boardImages')
```
queryBuilder 의 join 문의 첫번째 parameter 는 string | Entity 입니다.
string 을 사용할 경우 entity 가 수정됐을 때 string 도 함께 바꿔줘야하는 공수가 생길 수 있습니다.
Entity 를 넣었을 경우 Entity 클레스에서 정의한 정보를 토대로 typeorm 에서 매핑을 자동으로 해줍니다.
```ts
leftJoinAndSelect('board.user', 'user') === leftJoinAndSelect(User, 'user')
```

추가로 leftJoin 과 leftJoinAndSelect 에 차이가 있으니 한번 살펴보고 적절한 메서드를 사용하면 더 좋을거같아요